benchmarking/evaluations/metrics/priority_satisfaction.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+from api.ai.priority_algorithm.interfaces import Priority
+from api.models.team import Team
+from api.models.team_set import TeamSet
+from benchmarking.evaluations.interfaces import TeamSetMetric
+
+
+class PrioritySatisfaction(TeamSetMetric):
+    """
+    Calculates the number of priorities satisfied by the algorithm
+    """
+
+    def __init__(self, priorities: List[Priority], is_linear: bool, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.priorities = priorities
+        self.is_linear = is_linear
+
+    def calculate(self, team_set: TeamSet) -> float:
+        weights = (
+            self.compute_linear_weights()
+            if self.is_linear
+            else self.computer_exponential_weights()
+        )
+        total_score = 0
+        for team in team_set.teams:
+            priorities = self.priorities_satisfied(team)
+            product = sum(w * p for w, p in zip(weights, priorities))
+            total_score += product
+        # Normalize the scores between 0 and 1 for easier comparison
+        if self.is_linear:
+            total_score /= team_set.num_teams * sum(range(1, len(self.priorities) + 1))
```

Comment:
ngl I have no idea how this normalizes, might just be sleepy opey doing bad math but code seems slightly unintuitive to me

Maybe it would help to have a `theoretical_max` variable defined above and then this like is just `total_score / theoretical_max`